pkg/svc/workflow.go

```diff
@@ -75,12 +47,46 @@ func (s *workflowsrvc) Register(ctx context.Context, w *workflow.Workflow) (res
 	return s.store.AddWorkflow(ctx, w)
 }
 
-// Retrieve a Workflow from FuseML.
+// Get a Workflow.
 func (s *workflowsrvc) Get(ctx context.Context, w *workflow.GetPayload) (res *workflow.Workflow, err error) {
+	s.logger.Print("workflow.get")
 	return s.getWorkflow(ctx, w.Name)
 }
 
-// Assign a Workflow to a Codeset
+// Delete a Workflow and its assignments.
+func (s *workflowsrvc) Delete(ctx context.Context, d *workflow.DeletePayload) (err error) {
+	s.logger.Print("workflow.delete")
+	if _, err := s.getWorkflow(ctx, d.Name); err != nil {
+		s.logger.Print(err)
+		return err
+	}
+
+	// unassign all assigned codesets, if there's any
+	assignedCodesets := s.store.GetAssignedCodesets(ctx, d.Name)
+	for _, ac := range assignedCodesets {
+		err := s.unassignCodesetFromWorkflow(ctx, d.Name, ac.Codeset)
+		if err != nil {
+			return err
+		}
+	}
+
+	// delete tekton pipeline
+	err = s.backend.DeleteWorkflow(ctx, s.logger, d.Name)
+	if err != nil {
+		s.logger.Print(err)
+		return err
+	}
+
+	// delete workflow
+	err = s.store.DeleteWorkflow(ctx, d.Name)
```

Comment:
I must have missed this earlier. Why explicitly call _store_ and _backend_ from _service_? What about just calling store, which would take care of backend too?

Reply:
workflow store does not have access to the backend.

Reply:
Right, and I suggested it could have :-) Therefor you would not need to access both store and backend from the service, only the store.

As I wrote, I probably missed some previous discussion about this.

Reply:
The agreement was to revisit this after MVP and the probable solution is to create a higher component that uses/hides both the workflow backend and the workflow store components. I tried the idea of using store as a front-end for everything in https://github.com/fuseml/fuseml-core/pull/50, but I agree that the store component only provides half of the functionality: persistent storage for the fuseml state related to workflows.

Reply:
OK, let's revisit it later, it's obviously bigger topic